### PR TITLE
Toys and stuff

### PR DESCRIPTION
**Pull request by JStheguy**
-Adds another foam dart gun based on the double barrled shotgun.
-Cap Gun, Foam Dart Crossbow, Foam Dart Shotgun have their own sprites.
-Foam Dart crossbow now only has a dart capacity of 1.
-Foam dart guns now come in boxes with extra darts.
-Adds Spare Foam Dart box.
-Adds 36 action figures sprited by MrSnapwalk. (Arcade prizes)
-And the foam dart/foam dart gun box things are now arcade prizes in
place of the plain foam dart guns.

### CODE REVIEW

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+ * Action Figures
+ */
 
+/obj/item/toy/figure
+	name = "Non-Specific Action Figure action figure"
+	desc = "A \"Space Life\" brand... wait, what the hell is this thing?"
+	icon_state = "nuketoy"
+	var/cooldown = 0
+	var/toysay = "What the fuck did you do?"
+	var/obj/item/toy/accessory/hastoy = null
```
Comment (Iamgoofball) [inline code/game/objects/items/toys.dm:949]: what's going on with this?

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
 
+/obj/item/toy/figure/attack_self(mob/user as mob)
+	if(!cooldown)
+		user << "<span class='notice'>The [src] says \"[toysay]\".</span>"
```
Comment (Iamgoofball) [inline code/game/objects/items/toys.dm:953]: it should say this to everyone in the area.

You should also be able to rename the toys for shits and giggles.

Comment (Atskadan) [inline code/game/objects/items/toys.dm:953]: no

**Comment by JStheguy** [comment]
Okay it should work now.

File: code/game/objects/items/toys.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/game/objects/items/toys.dm:953]: Why not?

**Comment by RemieRichards** [comment]
no... these are not toy figures of crew/syndies...

WOW. want.

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Chief Engineer action figure"
+	desc = "A \"Space Life\" brand Chief Engineer action figure."
+	icon_state = "ce"
+	toysay = "Oh god the singularity is loose!"
+
+/obj/item/toy/figure/chaplain
+	name = "Chaplain action figure"
+	desc = "A \"Space Life\" brand Chaplain action figure."
+	icon_state = "chaplain"
+	toysay = "Praise Space Jesus!"
```
Comment (RemieRichards) [inline code/game/objects/items/toys.dm:1026]: could actually have this check against the current round's deity, and if there isn't one, set it to "Space Jesus" as default.

**Comment by MrSnapwalk** [comment]
Here, have a picture of all the action figures. These were fun to make!

**Comment by RemieRichards** [comment]
@MrSnapwalk  Hnnnnng OMG they are SOOOO CUTE

**Comment by JStheguy** [comment]
Read: I'm too fucking lazy at this point to try and figure out how fucking chaplain deity shit works nor try to figure out how to make that shit change what the action figure says.

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Botanist action figure"
+	desc = "A \"Space Life\" brand Botanist action figure."
+	icon_state = "botanist"
+	toysay = "Dude, I see colors..."
+
+/obj/item/toy/figure/captain
+	name = "Captain action figure"
+	desc = "A \"Space Life\" brand Captian action figure."
+	icon_state = "captain"
+	toysay = "Any Heads of Staff?"
```
Comment (MrSnapwalk) [inline code/game/objects/items/toys.dm]: Should be "Any heads of staff?".

File: code/game/objects/items/toys.dm [hunk 5]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Chaplain action figure"
+	desc = "A \"Space Life\" brand Chaplain action figure."
+	icon_state = "chaplain"
+	toysay = "Join my religion, you fucks."
+
+/obj/item/toy/figure/chef
+	name = "Chef action figure"
+	desc = "A \"Space Life\" brand Chef action figure."
+	icon_state = "chef"
+	toysay = "Bartender is a tastey burger."
```
Comment (MrSnapwalk) [inline code/game/objects/items/toys.dm]: You misspelled tasty in this line.

File: code/game/objects/items/toys.dm [hunk 6]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Virologist action figure"
+	desc = "A \"Space Life\" brand Virologist action figure."
+	icon_state = "virologist"
+	toysay = "The cure is potassium!"
+
+/obj/item/toy/figure/warden
+	name = "Warden action figure"
+	desc = "A \"Space Life\" brand Warden action figure."
+	icon_state = "warden"
+	toysay = "17 Minutes for coughing at an officer!"
```
Comment (MrSnapwalk) [inline code/game/objects/items/toys.dm]: "Minutes" should be lower case.

**Comment by MrSnapwalk** [comment]
Pointed out some spelling and capitalization mistakes, it's up to you if you feel like fixing them. Either way, good job! Looking forward to having these in-game.

File: code/game/objects/items/toys.dm (hunk 6 above)
Comment (Atskadan) [inline code/game/objects/items/toys.dm]: Numbers which start a sentence are always spelled out, it should be "Seventeen"

File: code/game/objects/items/toys.dm (hunk 5 above)
Comment (Atskadan) [inline code/game/objects/items/toys.dm]: This line doesn't make sense either

File: code/game/objects/items/toys.dm (hunk 2 above)
Comment (Atskadan) [inline code/game/objects/items/toys.dm:953]: no

File: code/game/objects/items/toys.dm [hunk 7]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Bartender action figure"
+	desc = "A \"Space Life\" brand Bartender action figure."
+	icon_state = "bartender"
+	toysay = "Where is Pun Pun?"
+
+/obj/item/toy/figure/borg
+	name = "Cyborg action figure"
+	desc = "A \"Space Life\" brand Cyborg action figure."
+	icon_state = "borg"
+	toysay = "I live again."
```
Comment (Atskadan) [inline code/game/objects/items/toys.dm]: all caps pls
and periods after every word

File: code/game/objects/items/toys.dm [hunk 8]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Chief Engineer action figure"
+	desc = "A \"Space Life\" brand Chief Engineer action figure."
+	icon_state = "ce"
+	toysay = "Oh god the singularity is loose!"
+
+/obj/item/toy/figure/chaplain
+	name = "Chaplain action figure"
+	desc = "A \"Space Life\" brand Chaplain action figure."
+	icon_state = "chaplain"
+	toysay = "Join my religion, you fucks."
```
Comment (Atskadan) [inline code/game/objects/items/toys.dm]: something else please like
"Glory to X" X being the deity

File: code/game/objects/items/toys.dm [hunk 9]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Clown action figure"
+	desc = "A \"Space Life\" brand Clown action figure."
+	icon_state = "clown"
+	toysay = "Honk!"
+
+/obj/item/toy/figure/ian
+	name = "Ian action figure"
+	desc = "A \"Space Life\" brand Ian action figure."
+	icon_state = "ian"
+	toysay = "Aruuu!"
```
Comment (Atskadan) [inline code/game/objects/items/toys.dm]: ian isn't a wolf
he says AUUUU or ARF
arf would be better

File: code/game/objects/items/toys.dm [hunk 10]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+
+/obj/item/toy/figure/detective
+	name = "Detective action figure"
+	desc = "A \"Space Life\" brand Detective action figure."
+	icon_state = "detective"
+	toysay = "The airlock has grey jumpsuit and insulated glove fibers on it."
```
Comment (Atskadan) [inline code/game/objects/items/toys.dm]: THIS airlock

File: code/game/objects/items/toys.dm [hunk 11]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Cargo Technician action figure"
+	desc = "A \"Space Life\" brand Cargo Technician action figure."
+	icon_state = "cargotech"
+	toysay = "For Cargonia!"
+
+/obj/item/toy/figure/ce
+	name = "Chief Engineer action figure"
+	desc = "A \"Space Life\" brand Chief Engineer action figure."
+	icon_state = "ce"
+	toysay = "Oh god the singularity is loose!"
```
Comment (Atskadan) [inline code/game/objects/items/toys.dm]: why not give this to the engineer and change it to "Wire the solars!"

File: code/game/objects/items/toys.dm [hunk 12]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Head of Security action figure"
+	desc = "A \"Space Life\" brand Head of Security action figure."
+	icon_state = "hos"
+	toysay = "Get the justice chamber ready, I think we got a joker here."
+
+/obj/item/toy/figure/qm
+	name = "Quartermaster action figure"
+	desc = "A \"Space Life\" brand Quartermaster action figure."
+	icon_state = "qm"
+	toysay = "Please sign this form in triplicate and we will see about geting you a welding mask within 3 buisness days."
```
Comment (Atskadan) [inline code/game/objects/items/toys.dm]: misspelled business

File: code/game/objects/items/toys.dm [hunk 13]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Janitor action figure"
+	desc = "A \"Space Life\" brand Janitor action figure."
+	icon_state = "janitor"
+	toysay = "Look at the signs, you idiot."
+
+/obj/item/toy/figure/lawyer
+	name = "Lawyer action figure"
+	desc = "A \"Space Life\" brand Lawyer action figure."
+	icon_state = "lawyer"
+	toysay = "My client is not guilty!"
```
Comment (Atskadan) [inline code/game/objects/items/toys.dm]: no no make it OBJECTION

File: code/game/objects/items/toys.dm [hunk 14]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	toysay = "Blowing all of the borgs!"
+
+/obj/item/toy/figure/roboticist
+	name = "Roboticist action figure"
+	desc = "A \"Space Life\" brand Roboticist action figure."
+	icon_state = "roboticist"
+	toysay = "Big stompy mechs!"
+
+/obj/item/toy/figure/scientist
+	name = "Roboticist action figure"
```
Comment (Atskadan) [inline code/game/objects/items/toys.dm]: should be scientist

File: code/game/objects/items/toys.dm (hunk 1 above)
Comment (Neerti) [inline code/game/objects/items/toys.dm:949]: it's an abstract, for inheritance.  It's not meant to be seen by players normally.

File: code/game/objects/items/toys.dm (hunk 8 above)
Comment (RemieRichards) [inline code/game/objects/items/toys.dm]: I actually removed the global deity variable when I cleaned up Bible vars sitting on the damn Game ticker.

I'd suggest

```
var/deity = pick(bibledeitynames) //bibledeitynames IS the name of the random deity list.
toysay = "Glory to [deity]!"
```

File: code/game/objects/items/toys.dm [hunk 15]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Botanist action figure"
+	desc = "A \"Space Life\" brand Botanist action figure."
+	icon_state = "botanist"
+	toysay = "Dude, I see colors..."
+
+/obj/item/toy/figure/captain
+	name = "Captain action figure"
+	desc = "A \"Space Life\" brand Captian action figure."
+	icon_state = "captain"
+	toysay = "Any heads of staff?"
```
Comment (ghost) [inline code/game/objects/items/toys.dm:1007]: I had quite the giggle, most of these are p good.

File: code/game/objects/items/toys.dm [hunk 16]
```diff
@@ -939,8 +936,246 @@ obj/item/toy/cards/deck/syndicate
...
+	name = "Chaplain action figure"
+	desc = "A \"Space Life\" brand Chaplain action figure."
+	icon_state = "chaplain"
+	toysay = "Join my religion, you fucks."
+
+/obj/item/toy/figure/chef
+	name = "Chef action figure"
+	desc = "A \"Space Life\" brand Chef action figure."
+	icon_state = "chef"
+	toysay = "Bartender is a tasty burger."
```
Comment (ChangelingRain) [inline code/game/objects/items/toys.dm]: You, uh, do mean Pun-Pun right?

**Comment by Atskadan** [comment]
pls update

**Comment by JStheguy** [comment]
Why update?

**Comment by Atskadan** [comment]
uh.. so that i'll merge it?

**Comment by JStheguy** [comment]
Srs what is there to update.

**Comment by Atskadan** [comment]
there's still an error in one of your descriptions and i would like (but not require) that you make the changes i suggested as well

**Comment by RemieRichards** [comment]
you saw nothing.

**Comment by Atskadan** [comment]
remie it can be merged

**Comment by RemieRichards** [comment]
Oh right.

**Comment by Atskadan** [comment]
i saw absolutely nothing.

File: code/game/objects/items/toys.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/game/objects/items/toys.dm:953]: yes

**Comment by JStheguy** [comment]
Yeah is good.

**Comment by JStheguy** [comment]
OKAY YEAH IS GOOD TO GO NOW(?)

**Comment by Atskadan** [comment]
no you still have the thing with the chaplain quote
remie even told you how to do it

**Comment by JStheguy** [comment]
FISSION MAILED

**Comment by JStheguy** [comment]
Yeah I cant get this shit to work.

**Comment by Atskadan** [comment]
are you getting any compile errors or anything?

**Comment by JStheguy** [comment]
Yes.
code/game/objects/items/toys.dm:1026:error: toydeity: invalid variable
code/game/objects/items/toys.dm:1026:error: : expected a constant expression

(Toydeity being just the deity thing from remies original snippit of code)
(and line 1026 being the chaplains Toysay line.)

(Right now the current code on my branch no longer has the deity whatthefuck)

**Comment by Iamgoofball** [comment]
a var defined like that can't be random on definiton

it needs to be defined and THEN made random on the new:

SO:

/obj/dick/butt
     desc = "hm, goku is a"
     var/dickinbutt = ""

/obj/dick/butt/New()
     ..()
     dickinbutt = "[pick(bibledeitynames)]"
     desc = "hm, goku is a [dickinbutt]"

is what works, as a variable can't be randomized on definition like you had it before.

**Comment by JStheguy** [comment]
Okay is good now?
If so you can merge.

**Pull request merged**